unyt/dask_array.py

```diff
@@ -0,0 +1,620 @@
+
+    @_special_dec
+    def __pow__(self, other):
+        return super().__pow__(other)
+
+    @_special_dec
+    def __mul__(self, other):
+        return super().__mul__(other)
+
+    @_special_dec
+    def __rmul__(self, other):
+        return super().__rmul__(other)
+
+    @_special_dec
+    def __truediv__(self, other):
+        return super().__truediv__(other)
+
+    @_special_dec
+    def __rtruediv__(self, other):
+        return super().__rtruediv__(other)
+
+    @_special_dec
+    def __add__(self, other):
+        return super().__add__(other)
+
+    @_special_dec
+    def __sub__(self, other):
+        return super().__sub__(other)
+
+    @_special_dec
+    def __lt__(self, other):
+        return super().__lt__(other)
+
+    @_special_dec
+    def __le__(self, other):
+        return super().__le__(other)
+
+    @_special_dec
+    def __gt__(self, other):
+        return super().__gt__(other)
+
+    @_special_dec
+    def __ge__(self, other):
+        return super().__ge__(other)
+
+    @_special_dec
+    def __eq__(self, other):
+        return super().__eq__(other)
+
+    @_special_dec
+    def __ne__(self, other):
+        return super().__ne__(other)
+
+    def prod(self, *args, **kwargs):
+        _, unit = ua._apply_power_mapping(
+            np.multiply, self._unyt_array.units, self.size, self.shape, kwargs
+        )
+        dask_result = super().prod(*args, **kwargs)
+        return _create_with_quantity(dask_result, ua.unyt_array([1.0], unit))
+
+
+def _finalize_unyt(results, unit_name):
+    """
+    the function to call from the __dask_postcompute__ hook.
+
+    Parameters
+    ----------
+    results : the dask results object
+    unit_name : the units of the result
+
+    Returns
+    -------
+    unyt_array or unyt_quantity
+
+    """
+
+    # here, we first call the standard finalize function for a dask array
+    # and then return a standard unyt_array from the now in-memory result if
+    # the result is an array, otherwise return a unyt_quantity.
+    result = _dask_finalize(results)
+
+    if type(result) == np.ndarray:
+        return ua.unyt_array(result, unit_name)
+    else:
+        return ua.unyt_quantity(result, unit_name)
+
+
+def _create_with_quantity(dask_array, new_unyt_array):
+    """
+    this function instantiates a new unyt_dask_array instance and then sets
+    the unit state, including units. Used to wrap dask operations
+
+    Parameters
+    ----------
+    dask_array : a standard dask array
+    new_unyt_array : a standard unity array
+    remaining arguments get passed to unyt.unyt_array, check there for a
+    description.
+    """
+    out = unyt_from_dask(dask_array)
+
+    # attach the unyt_array
+    units = new_unyt_array.units
+    unyt_name = new_unyt_array.name
+
+    out._set_unit_state(units, new_unyt_array, unyt_name)
+    return out
+
+
+def unyt_from_dask(
+    dask_array,
+    units=None,
+    registry=None,
+    bypass_validation=False,
+    unyt_name=None,
```

Comment:
```suggestion
    *,
    registry=None,
    bypass_validation=False,
    unyt_name=None,
```
see https://github.com/yt-project/yt/issues/3992 if you need an explanation/rationale